### ISSUE TITLE
[Feature]: Robustness transfer learning - pre-trained adversarial representations as foundation for downstream tasks

### ISSUE THREAD

**Issue by Adit-Jain-srm** [Adit-Jain-srm/NightmareNet#31]
## Motivation

NightmareNet produces hardened models that are measurably more robust. But that robustness is task-specific: train on SST-2, get a robust SST-2 classifier. There's no mechanism to TRANSFER the learned robustness representations to a new task without re-running the full 4-phase cycle.

The hypothesis (supported by Salman et al. 2020): adversarially trained models learn fundamentally different internal representations that capture semantically meaningful features. These should transfer better to new tasks.

## The Gap

- User runs 3 cycles on SST-2 (30 min) -> robust SST-2 model
- User wants robust AG News classifier -> must run 3 cycles from scratch (another 30 min)
- The SST-2 robustness training is "wasted" for the new task

What SHOULD happen: fine-tune the robust SST-2 backbone on AG News (5 min) and get a model that's already partially robust without nightmare training on AG News.

### 1. Robustness Pre-training Registry

After a full cycle, save the backbone (everything except classification head) as a "robust foundation model." Metadata includes: training task, robustness score, cycles completed, distortion types survived.

### 2. Transfer Fine-tuning Pipeline

YAML config specifying: foundation model ID, target task/dataset, fine-tuning params (epochs, lr, freeze layers), and whether to run a distortion sweep on the result for comparison.

### 3. Transfer Robustness Measurement

Key metric: `Transfer Ratio = robustness(transferred) / robustness(full-cycle)`

- Ratio > 0.7: transfer is efficient (saves 70%+ of compute)
- Ratio < 0.3: transfer is weak (full cycle still needed)

### 4. CLI Extension

- `nightmarenet foundation register --model ./runs/sst2-v1 --name robust-distilbert-sst2`
- `nightmarenet transfer --foundation robust-distilbert-sst2 --config configs/transfer_ag_news.yaml`
- `nightmarenet transfer --measure --transferred ./result --baseline ./full_cycle_result`

### 5. Module Structure

New package `nightmarenet/transfer/` with:
- `registry.py` - Foundation model storage and retrieval
- `fine_tune.py` - Transfer fine-tuning with optional layer freezing
- `measurement.py` - Transfer ratio computation
- `head_factory.py` - Task-specific heads for classification, NER, generation
- `report.py` - Transfer efficiency report

### 6. Research Contribution

If transfer ratios consistently > 0.6 across task pairs, this validates NightmareNet's core claim: the 4-phase cycle learns genuinely robust representations, not task-specific hacks. Include as Section 5 in `docs/research/paper-draft.md`.

## Technical Considerations

- Foundation models stored WITHOUT classification head (backbone + embeddings only)
- Head factory creates appropriate layers based on task type
- Layer freezing: gradual unfreeze strategy (freeze bottom N for first epoch)
- Baseline comparison uses IDENTICAL hyperparameters for fair measurement
- Memory: foundation + new head should not exceed original model memory

## Acceptance Criteria

- [ ] Foundation registry saves/loads backbone models without task heads
- [ ] Transfer fine-tuning works from YAML config
- [ ] Transfer ratio measurement compares transferred vs full-cycle
- [ ] CLI commands: `foundation register`, `transfer`, `transfer --measure`
- [ ] At least one measured experiment (SST-2 -> AG News)
- [ ] Results documented in paper draft
- [ ] Layer freezing strategy configurable
- [ ] Head factory supports classification and token-classification
- [ ] Transfer report includes compute savings
- [ ] Tests: registry CRUD, head factory, measurement math

**Comment by Diwakar-odds** [comment]
Hi, I would like to work on this issue. Please assign this to me under ECSoC26.

**Comment by Adit-Jain-srm** [comment]
Sure @Diwakar-odds
